### PR TITLE
Better Error Tracking

### PR DESCRIPTION
**Pull request by ravigadad**
Collect errors from mergefields not interpolating, invalid formulas, missing data, etc on each mergefield, and assemble an error report to include in the generation exception.  Also refactor and rename some errors.

### CODE REVIEW

File: lib/sheng/docx.rb [hunk 1]
```diff
@@ -5,7 +5,9 @@
 module Sheng
   class Docx
     class InvalidFile < StandardError; end
+    class TemplateError < StandardError; end
     class FileAlreadyExists < StandardError; end
+    class GenerateError < StandardError; end
```
Comment (hungryed) [inline lib/sheng/docx.rb]: should we make these into one base error?

something like
```ruby
class FIleError < StandardError; end
class InvalidFile < FIleError; end
```

Comment (ravigadad) [inline lib/sheng/docx.rb]: Hm.  I see your intention here, but I'm wondering how to create a grouping that would actually be useful.  `InvalidFile` is an unreadable zip file; `TemplateError` is any issue with the mergefield arrangement that could be found through parsing the structure (broken sequences, etc), and `GenerateError` is any errors thrown *while* merging.  `FileAlreadyExists` is actually thrown when the output path holds an existing file.

Comment (ravigadad) [inline lib/sheng/docx.rb]: Let me mess around with some rename/subclassing possibilities and see if I can come up with something reasonable for clients

**Comment by ravigadad** [comment]
@hungryed:  Made some changes to error classes; please take a look

**Comment by hungryed** [comment]
👍

**Pull request merged**